src/components/MessagePanel/message-panel.jsx

```diff
@@ -74,32 +75,37 @@ class MessagePanel extends Component {
     event.source.postMessage(msgStructure, event.origin);
   }
 
-  addMessage(event) {
-    if (event.origin.includes(document.domain)) {
-      console.log(`Received message from ${event.origin} but it is the same as the current document's domain: ${document.domain}.`);
-      return;
-    }
-
-    console.log(`Received message from ${event.origin}.`);
+  isActionableMessage(event) {
+    // Ignore the event, if it doesn't meet our expectations.
+    if (!event.data) return false;
+    if ((event.data.source || "").startsWith('react-devtools-')) return false;
 
     if (!event.data.messageId) {
-      console.log('Message did not have a messageId and will be ignored.');
-      return;
+      console.warn('Message has no messageId.');
+      return false;
     }
 
     if (!event.data.messageType) {
-      console.log('Message did not have a messageType and will be ignored.');
-      return;
+      console.warn('Message has no messageType.');
+      return false;
     }
 
-    if (!(event.data.messageType.includes('scratchpad.') || event.data.messageType.includes('ui.'))) {
-      console.log('Message did not have a supported messageType and will be ignored.');
-      return;
+    const messageTarget = event.data.messageType.split('.');
+    if (!['scratchpad', 'ui'].includes(messageTarget[0])) {
+      console.warn(`Unknown message type '${messageTarget[0]}.`);
+      return false;
     }
 
-    const message = JSON.stringify(event.data, null, 2);
-    this.setState({ messages: [...this.state.messages, message] });
-    this.replyMessage(event);
+    return true;
+  }
+
+  addMessage(event) {
+    if (this.isActionableMessage(event)) {
+      const message = JSON.stringify(event.data, null, 2);
+      // TODO: convert this to a stack so newer messages are on top.
+      this.setState({ messages: [...this.state.messages, message] });
```

Comment:
Do you want to swap the order of array elements here, so the splice comes in later? Otherwise, maybe remove the TODO or convert to a github issue.

```suggestion
      this.setState({ messages: [message, ...this.state.messages] });
```
... but this probably needs additional tests, too. So, your call on how to incorporate.

Reply:
I've created a separate issue here: https://github.com/cds-hooks/sandbox/issues/129